applications/crossbar/test/cb_storage_test.erl

```diff
@@ -20,10 +64,113 @@ maybe_check_storage_settings_test_() ->
+     ,?_assertEqual(400, cb_context:resp_error_code(PutRespContext))
+     }
+    ,{"If storage settings validation fails, crossbar should return an error message"
+     ,?_assertEqual(<<"validation error">>, cb_context:resp_error_msg(PutRespContext))
+     }
+    ,{"If you provide invalid credentials for S3 it should return 403"
+     ,?_assertEqual(403, s3_error_code(UUID, PutRespContext))
+     }
+    ,{"If you provide invalid credentials for S3 it should return an error message"
+     ,?_assertEqual(<<"The AWS Access Key Id you provided does not exist in our records.">>,
+                    s3_message(UUID, PutRespContext))
+     }
     ].
 
 %% =======================================================================================
 %% Helper functions
 %% =======================================================================================
+s3_storage_plan(UUID) ->
+    #{<<"attachments">> =>
+          #{UUID =>
+                #{<<"handler">> => <<"s3">>
+                 ,<<"name">> => <<"S3 Storage">>
+                 ,<<"settings">> =>
+                      #{<<"bucket">> => <<"my_S3_BUCKET">>
+                       ,<<"key">> => <<"my_AWS_ACCESS_KEY">>
+                       ,<<"secret">> => <<"my_AWS_SECRET_KEY">>
+                       }
+                 }
+           }
+     ,<<"id">> => ?ACCOUNT_ID_UNDER_TEST
+     ,<<"plan">> =>
+          #{<<"modb">> =>
+                #{<<"types">> =>
+                      #{<<"mailbox_message">> =>
+                            #{<<"attachments">> =>
+                                  #{<<"handler">> => UUID
+                                   }
+                             }
+                       }
+                 }
+           }
+     }.
+
 maybe_check(Context, HTTPVerb) ->
     cb_storage:maybe_check_storage_settings(Context, HTTPVerb).
+
+%% this is required for voice mail messages which their timestamp is missing from metadata and
+%% their private media is missing or both private media and mailbox doesn't have create/modified
+this_month_db_exists() ->
+    fun(Server, Db) ->
+            %% why meck can't call the original mfa properly?
+            %% meck_code_gen:get_current_call/1 returns undefined and it cause bad_match
+            %% in meck:passthrough/1
+            (kz_datamgr:db_classification(Db) =:= 'modb'
+             andalso is_db_under_test(Db, kazoo_modb:get_modb(kz_util:format_account_id(Db))))
+                orelse erlang:apply('kz_fixturedb_db_meck_original', db_exists, [Server, Db])
+    end.
+
+is_db_under_test(ThisMonth, ThisMonth) ->                                             true;
+is_db_under_test(<<"account%2Fac%2Fco%2Funt0000000000000000000000001-201710">>, _) -> true;
+is_db_under_test(_, _) ->                                                             false.
+
+-define(GET_LEGACY_CALL, {kz_datamgr, get_results, [?ACCOUNT_DB_UNDER_TEST, ?LEGACY_VIEW, [{limit, 2000}, descending]]}).
```

Comment:
This is for kazoo_voicemail migration test, remove it, also remove `check_kz_datamgr_history` since you don't need to meck `get_result` function.